### ISSUE TITLE
ThreadSafeReference & readonly-configuration crashes

### ISSUE THREAD

**Issue by haemi** [realm/realm-cocoa#5475]
when I configure my realm as `readonly`, the app crashes whenever I try to create a `ThreadSafeReference(to: ...)` instance. The breakpoint stays here:

```
thread_safe_reference.cpp#39:
m_version_id = get_source_shared_group().pin_version();
```

Without `readonly:true` it works. But as I'm using a realm from the bundle I need `readonly`.

RealmSwift 3.0.2, iOS 11

**Comment by kishikawakatsumi** [comment]
Thank you for reporting. We will fix it soon.

**Comment by phuocmobytelab** [comment]
hi all, do u fix this bug? i met the same issue @haemi

**Comment by DominicFrei** [comment]
I created a PR for this issue in core: https://github.com/realm/realm-core/pull/4517

**Comment by DominicFrei** [comment]
The PR mentioned above is now on hold since we need to do some re-works first to the read-only realms: https://github.com/realm/realm-core/issues/4528

**Comment by DominicFrei** [comment]
TL;DR Fixed in Core and will be released soon.

Update:
The above mentioned https://github.com/realm/realm-core/issues/4528 which emerged out of this issue was merged in core. Subsequently https://github.com/realm/realm-core/pull/4517 was closed and https://github.com/realm/realm-core/pull/4688 opened instead since the solution looks a bit different now thanks to the first mentioned PR.
This needs to be released in core and then subsequently included and released in Cocoa.

**Comment by tgoyne** [comment]
This is fixed in 10.7.6.
